src/front/mod.rs

```diff
@@ -89,6 +89,19 @@ impl Typifier {
         self.resolutions[expr_handle.index()].inner_with(types)
     }
 
+    pub fn get_handle(
+        &self,
+        expr_handle: Handle<crate::Expression>,
+        types: &mut UniqueArena<crate::Type>,
+    ) -> Handle<crate::Type> {
```

Comment:
Rather than doing this, I think we should use `TypeResolution` more places in `front::wgsl`, and just lean on the `ensure_type_exists` methods to convert those into `Handle` values when we really need to. Working on a patch for this now.